mlir/lib/Interfaces/DataLayoutInterfaces.cpp

```diff
@@ -324,6 +346,19 @@ static DataLayoutSpecInterface getSpec(Operation *operation) {
       });
 }
 
+static TargetSystemSpecInterface getTargetSystemSpec(Operation *operation) {
+  if (operation) {
+    ModuleOp moduleOp;
+    if (isa<ModuleOp>(operation)) {
+      moduleOp = llvm::dyn_cast<ModuleOp>(operation);
+    } else {
+      moduleOp = operation->getParentOfType<ModuleOp>();
+    }
+    return moduleOp.getTargetSystemSpec();
+  } else
```

Comment:
No else after return 

```suggestion
  }
```